.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,328 @@
+name: Build and Publish NuGet Package for all supported platforms
+
+on:
+  push:
+  pull_request:
+    branches:
+    - master
+
+
+env:
+  MAJOR: 4
+  MINOR: 0
+  RELEASE: 0
+  RUN: ${{ github.run_number }}
```

Comment:
let's increase the minor version to 1 (to make sure we don't get the "downgrade" from the current version 4.0.0.N to 4.0.0.1 or so)